File: android-app/app/src/main/java/com/pranavakshit/gpscamportal/data/local/PhotoDao.java
package com.pranavakshit.gpscamportal.data.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import java.util.List;

@Dao
public interface PhotoDao {
    @Insert
    void insertPhoto(PhotoEntity photo);

    @Query("SELECT * FROM photos ORDER BY timestamp DESC")
    LiveData<List<PhotoEntity>> getAllPhotos();

    @Query("SELECT * FROM photos WHERE isUploaded = 0 ORDER BY timestamp DESC")
    LiveData<List<PhotoEntity>> getPendingUploads();

    @Query("UPDATE photos SET isUploaded = 1 WHERE id IN (:photoIds)")
    void markAsUploaded(List<Integer> photoIds);

    @Query("DELETE FROM photos WHERE id = :id")
    void deletePhoto(int id);
}

File: android-app/app/src/main/java/com/pranavakshit/gpscamportal/data/local/PhotoDao.kt
package com.pranavakshit.gpscamportal.data.local

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import kotlinx.coroutines.flow.Flow

@Dao
interface PhotoDao {
    @Insert
    suspend fun insertPhoto(photo: PhotoEntity)

    @Query("SELECT * FROM photos ORDER BY timestamp DESC")
    fun getAllPhotos(): Flow<List<PhotoEntity>>

    @Query("SELECT * FROM photos WHERE isUploaded = 0 ORDER BY timestamp DESC")
    fun getPendingUploads(): Flow<List<PhotoEntity>>

    @Query("UPDATE photos SET isUploaded = 1 WHERE id IN (:photoIds)")
    suspend fun markAsUploaded(photoIds: List<Int>)

    @Query("DELETE FROM photos WHERE id = :id")
    suspend fun deletePhoto(id: Int)
}
